cmd/tools/gendynamicconfig/main.go

```diff
@@ -95,8 +95,8 @@ var (
 		},
 		{
 			Name:   "NamespaceID",
-			GoArgs: "namespaceID string",
-			Expr:   "[]Constraints{{NamespaceID: namespaceID}, {}}",
+			GoArgs: "namespaceID namespace.ID",
+			Expr:   "[]Constraints{{NamespaceID: namespaceID.String()}, {}}",
```

Comment:
Did you consider changing the type of the field in the Constraints struct also?

_Technically_ this is public (i.e. through ServerOptions) api, so we shouldn't, but we could argue it's trivial and very likely no one would notice.

Reply:
I haven't considered that yet, I felt most of the value was captured by ensuring typesafety on the API.

I've got no strong opinion either way there.

But first, I was going to do the same for namespace name to wrap that up.